custom-pages/gadgets/reports/reports.js

```diff
@@ -1,13 +1,192 @@
 define(function (require, exports, module) {
-    
+    let XLSX = {}
+    let wb
+
+    //TODO reduce size of includes via lib
     require('css!./styles/reports.css');
+    require(['./lib/dist/xlsx.full.min.js'], function() {
+        //global function defined in dependency
+        make_xlsx_lib(XLSX) //eslint-disable-line no-undef
+    });
     const html = require('text!./templates/reports.html');
     
     const Empty = require('ratchet/dynamic/empty');
     
     const UI = require('ui');
-    
+    const $ = require('jquery')
+
+    const [PAGE, PRODUCT, SKU, PRICESHEET] = [
+        'page',
+        'product',
+        'sku',
+        'price-sheet'
+    ]
+    const ReportTypes = [PAGE, PRODUCT, SKU, PRICESHEET]
+
+    const QueryTypes = {
+        [PAGE]: {
+            "$regex": "cricket:page(-.+)?"
+        },
+        [PRODUCT]: "cricket:product",
+        [SKU]: "cricket:sku",
+        [PRICESHEET]: "cricket:price-list"
+    }
+
     let branch;
+    let platform;
+
+    function buildQuery(reportType) {
+        let query = { }
+        if (reportType in QueryTypes) {
+            query._type = QueryTypes[reportType]
+        } else {
+            console.error('Invalid report type provided')
+        }
+
+        if (query._type) {
+            return query
+        } else {
+            return false
+        }
+    }
+
+    function genericErrorLoggerHalter(err) {
+        console.error(err)
+        return false
+    }
+
+    function buildWorksheet({nodes, reportType}) {
+        const topLevelFieldsToCopy = ["title", "active", "sol", "soli", "eol", "eoli"]
+        let filteredNodes = []
+
+        nodes = nodes.map((record) => {
+            let rec = {}
+
+            //copy common fields
+            topLevelFieldsToCopy.forEach(function (fieldName) {
+                rec[fieldName] = record[fieldName]
+            })
+
+            rec._type = record.getTypeQName()
+            let meta = record.getSystemMetadata()
+            rec["_system.created_on.timestamp"] = meta.created_on.timestamp
```

Comment:
FYI,

you can also call meta.getCreatedOn()

Reply:
Have switched all of the meta methods now.